### ISSUE TITLE
Cometbft v0.37.5 is not compatible with CosmosSDK v0.47.10

### ISSUE THREAD

**Issue by dimiandre** [cometbft/cometbft#2831]
With cometbft v0.37.5 viper has been bumped to v1.18.1

this breaks custom app.toml configurations

In specific, in this function https://github.com/cosmos/cosmos-sdk/blob/d1b5b0c5ae2c51206cc1849e09e4d59986742cc3/server/util.go#L223 

customConfig is an interface, and viper is not able to unmarshal it because it expects a map https://github.com/cosmos/cosmos-sdk/blob/4912bef7fb14ebe5da96ed96e8c47b609506cf49/server/util.go#L275

```
1:49PM ERR failure when running app err="failed to parse /home/dimi/.juno/config/app.toml: '' expected a map, got 'interface'"
```

replacing viper in chain's go.mod fix this error 

```
replace github.com/spf13/viper => github.com/spf13/viper v1.16.0
```

**Comment by andynog** [comment]
Hi @dimiandre, thanks for opening this issue. As I understand, the issue can be fixed if you add that replace in the application go.mod. Since this is an issue in the SDK, should this be reported on their repo ? Is there anything do you need from CometBFT side ? (Or did you open the issue to make people aware that this might cause some issues with the SDK v0.37.5 version?)

**Comment by dimiandre** [comment]
seems to be fixed with sdk 47.11

thanks
